ops/pebble.py

```diff
@@ -114,19 +117,25 @@ def write(self, buf: Union[bytes, str, bytearray]) -> int: ...  # noqa
                                "group-id": Optional[int],
                                "group": Optional[str],
                                "type": Union['FileType', str]})
-    _HttpDict = TypedDict('_HttpDict', {'url': str})
-    _TcpDict = TypedDict('_TcpDict', {'port': int})
-    _ExecDict = TypedDict('_ExecDict', {'command': str})
-    _CheckDict = TypedDict('_CheckDict',
-                           {'override': str,
-                            'level': Union['CheckLevel', str],
-                            'period': Optional[str],
-                            'timeout': Optional[str],
-                            'http': Optional[_HttpDict],
-                            'tcp': Optional[_TcpDict],
-                            'exec': Optional[_ExecDict],
-                            'threshold': Optional[int]},
-                           total=False)
+
+    class _HttpDict(TypedDict):
+        url: str
+
+    class _TcpDict(TypedDict):
+        port: int
+
+    class _ExecDict(TypedDict):
+        command: str
+
+    class _CheckDict(TypedDict, total=False):
+        override: str
+        level: Union['CheckLevel', str]
+        period: Optional[str]
+        timeout: Optional[str]
+        http: Optional[_HttpDict]
+        tcp: Optional[_TcpDict]
+        exec: Optional[_ExecDict]
+        threshold: Optional[int]
 
     _AuthDict = TypedDict('_AuthDict',
```

Comment:
I really don't understand how it found all the others, but keeps missing some of them.
Is there a special field that isn't supported by the class syntax? (user-id looks suspicious). Hm, that makes me wonder if the new syntax is actually better, or whether it just sets you up to think of things in terms of a Type, when it really is just a dict. (it *is* far fewer quotes and commas, but that seems to be the only real difference)

Reply:
I haven't gone through an automated process that closely -- iterative improvements will take a bit, and it's easier (from my POV) to take a single "I ran this tool" PR and merge it into `main` with a smaller PR to reconcile what it may have missed later than to line-item them.

In general, tend to look at stuff like this and say "well, since Python is still pretty dependent on the exact ordering of class/variable definition in a single file in 2023, maybe converting this particular one to a class would result in some impossible intermediate bytecode". I don't know why a fully-completed class definition if (if this were just `class _AuthDict(TypedDict, ...`) wouldn't have finished bytecode evaluation by then, but it's sometimes surprisingly fragile.

It doesn't hurt to see what happens if it's replaced by hand, but I'd also expect the tool to be a little conservative.

Reply:
@jameinel See above: https://github.com/canonical/operator/pull/890/files/49e6c623ecc2802d141f4420ff8ffb30641b05aa#r1082094668